lib/bundler/definition.rb

```diff
@@ -904,7 +904,7 @@ def expand_dependencies(dependencies, remote = false)
             "To add those platforms to the bundle, " \
             "run `bundle lock --add-platform #{mapped_platforms.join " "}`."
         end
-        platforms.each do |p|
+        Resolver.sort_platforms(platforms).each do |p|
```

Comment:
I'm a little concerned about sorting in `definition`. Is this on the require path?
We should skip this block entirely if `remote == true` since the entire functionality is not used.